GitUI/UserControls/BlameControl.cs

```diff
@@ -457,33 +457,50 @@ private void copyCommitHashToClipboardToolStripMenuItem_Click(object sender, Eve
             CopyToClipboard(c => c.ObjectId.ToString());
         }
 
-        private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventArgs e)
+        private bool TryGetSelectedRevision(out GitRevision selectedRevision)
         {
+            selectedRevision = null;
             int line = (int?)contextMenu.Tag ?? -1;
             if (line < 0)
             {
-                return;
+                return false;
             }
 
             var objectId = _blame.Lines[line].Commit.ObjectId;
-            int originalLine = _blame.Lines[line].OriginLineNumber;
-            GitBlame blame = Module.Blame(_fileName, objectId + "^", _encoding, originalLine + ",+1");
-            if (blame.Lines.Count > 0)
+
+            selectedRevision = _revGrid.GetRevision(objectId);
```

Comment:
@mstv What is STR? Has Stuttgart Airport anything to do with this?

Reply:
No, there's no obvious relation ;-)
STR = steps to reproduce.

Reply:
This PR is not intended to fix that behavior that is caused by the regression about following renames. That's why I wrote :

>Partly fixes #6605 (other part will be fixed by #6807 or #6132)

And I also want one of these 2 PRs merged in the next release to have a consistent behavior.

Reply:
Yes, I agree. I had understood that it will need part 2. Though in order to test the disabling of the menu items, I found it helpful to have an example where it occurs.